### PR TITLE
Extract common server code to server-core.  Set up server in drophero

### PR DESCRIPTION
**Pull request by JustinB-SH**
**Description**
* Extracts common server code to `server-core` module
* Adds drophero project with running GRPC server.  Allows GRPC coroutine-based endpoints to run in a DropHero server.

### CODE REVIEW

File: kotlin/kroto-plus/server-core/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+dependencies {
+  api(project(":model"))
+  api("org.jetbrains.kotlinx", "kotlinx-coroutines-core")
+  implementation("javax.inject:javax.inject")
+}
+
+
```
Comment (adomokos) [inline kotlin/kroto-plus/server-core/build.gradle.kts]: Do you need these extra line breaks?

File: kotlin/kroto-plus/drophero/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    application
+}
+
+application {
+    mainClassName = "com.spothero.grpcjava.drophero.AppKt"
+}
+
+dependencies {
+    implementation(project(":server-core"))
+    implementation("com.spothero", "spothero-drophero", "0.5.2")
```
Comment (adomokos) [inline kotlin/kroto-plus/drophero/build.gradle.kts]: This is mostly for my own learning, but can express this the same way as:
`implementation("com.spothero:spothero-drophero:0.5.2")`

File: kotlin/kroto-plus/drophero/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    implementation("io.github.msteinhoff:dropwizard-grpc:1.2.3-2")
+
+    implementation(platform("com.fasterxml.jackson:jackson-bom:2.9.9"))
+    implementation("com.fasterxml.jackson.module", "jackson-module-kotlin")
```
Comment (adomokos) [inline kotlin/kroto-plus/drophero/build.gradle.kts]: I began looking at [klaxon](https://github.com/cbeust/klaxon), which seems like a Kotlin layer on top of Jackson.

File: kotlin/kroto-plus/server-core/build.gradle.kts (hunk 1 above)
Comment (JustinB-SH) [reply kotlin/kroto-plus/server-core/build.gradle.kts]: nope--removed

File: kotlin/kroto-plus/drophero/build.gradle.kts (hunk 2 above)
Comment (JustinB-SH) [reply kotlin/kroto-plus/drophero/build.gradle.kts]: yes.  We seem to have adopted the style of the 3-argument format in order service, so I maintained that here.

The 3-argument format is nice when you define the version in a property file.  You could use string interpolation with the property and use the single string.  But I find the 3-argument format a little easier to read.

**Pull request merged**